packages/babel-plugin-transform-typescript/src/enum.js

```diff
@@ -24,7 +24,9 @@ export default function transpileEnum(path, t) {
         path.remove();
       } else {
         const isGlobal = t.isProgram(path.parent); // && !path.parent.body.some(t.isModuleDeclaration);
-        path.replaceWith(makeVar(node.id, t, isGlobal ? "var" : "let"));
+        path.scope.registerDeclaration(
+          path.replaceWith(makeVar(node.id, t, isGlobal ? "var" : "let"))[0],
```

Comment:
Could you fix this in a separate PR, since it is an actual bugfix and thus less controversial? Probably all the `makeVar` calls need this fix.

Reply:
This is the only usage of the call.

Reply:
Additional reading: https://github.com/babel/babel/blob/9915c4f3b781483c18e118b6c5bfa4109453126b/packages/babel-plugin-transform-typescript/src/index.js#L70-L75

Reply:
#10034